pkg/kubelet/kubelet.go

```diff
@@ -2768,6 +2803,28 @@ func (kl *Kubelet) HandlePodUpdates(pods []*v1.Pod) {
 			}
 		}
 
+		if utilfeature.DefaultFeatureGate.Enabled(features.NodeDeclaredFeatures) {
+			oldPodInfo := &ndf.PodInfo{Pod: oldPod}
+			newPodInfo := &ndf.PodInfo{Pod: pod}
+			reqs, err := kl.nodeDeclaredFeaturesHelper.InferForPodUpdate(oldPodInfo, newPodInfo, kl.version)
+			if err != nil {
+				klog.ErrorS(err, "Failed to infer required features for pod update", "pod", klog.KObj(pod))
+			}
+			if len(reqs) != 0 {
+				matchResult, err := kl.nodeDeclaredFeaturesHelper.MatchCurrentNode(reqs, kl.nodeDeclaredFeatures)
+				if err != nil {
+					klog.ErrorS(err, "Failed to match pod features with the node", "pod", klog.KObj(pod))
+
+				}
+				if !matchResult.IsMatch {
+					missingNodeDeclaredFeatures := strings.Join(matchResult.UnsatisfiedRequirements, ", ")
+					klog.ErrorS(nil, "Pod requires node features that are not available", "missingFeatures", missingNodeDeclaredFeatures)
+					kl.recorder.Eventf(pod, v1.EventTypeWarning, events.FailedNodeDeclaredFeaturesCheck, "Pod requires node features that are not available: %s", missingNodeDeclaredFeatures)
+					// TODO(pravk03): Handle feature check failure. Currently, we just log an error and emit an event.
+				}
+			}
+		}
```

Comment:
I'm not sure we need to do this check here. We're already checking it at API admission time, so I think it's OK for the Kubelet to ignore the update check.

Reply:
removed.

Reply:
It's still here - did you forget to push the commit?